ethereum/blockcreation/src/main/java/org/hyperledger/besu/ethereum/blockcreation/txselection/selectors/AbstractStatefulTransactionSelector.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright contributors to Hyperledger Besu.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.ethereum.blockcreation.txselection.selectors;
+
+import org.hyperledger.besu.ethereum.blockcreation.txselection.BlockSelectionContext;
+import org.hyperledger.besu.plugin.services.txselection.SelectorsStateManager;
+
+/**
+ * This class represents an abstract transaction selector which provides methods to evaluate
+ * transactions.
+ */
+public abstract class AbstractStatefulTransactionSelector<S> extends AbstractTransactionSelector {
+  private final SelectorsStateManager selectorsStateManager;
+
+  public AbstractStatefulTransactionSelector(
+      final BlockSelectionContext context,
+      final SelectorsStateManager selectorsStateManager,
+      final S initialState,
+      final SelectorsStateManager.StateDuplicator<S> stateDuplicator) {
+    super(context);
+    this.selectorsStateManager = selectorsStateManager;
+    selectorsStateManager.createSelectorState(this, initialState, stateDuplicator);
+  }
+
+  protected S getWorkingState() {
+    return selectorsStateManager.getSelectorWorkingState(this);
+  }
+
+  protected void setWorkingState(final S newState) {
```

Comment:
not completly sure about this name, for me it was the WorldState and after I saw that it's a long 🤔

Reply:
This is about the selector specific state, for example the `BlobSizeTransactionSelector` that needs to keep the cumulative count of how many blobs have been already added by previous txs

Reply:
I don't know if it's better but maybe WorkingContext? SelectionContext?

Reply:
Not sure, we already have `BlockSelectionContext` and `TransactionEvaluationContext` here so it can also create confusion.
Do you think using _state_ could be confusing just at a first look, and after reviewing the code for longer the distinction is clearer?

I can also rename this method to `setSelectorWorkingState`, I thought it was redundant since it is in a selector class already, but if it helps readability then no problem

Reply:
I think it's ok, maybe just add a comment to explain what is this state

Reply:
done